public/app/features/trails/AutomaticMetricQueries/types.ts

```diff
@@ -0,0 +1,17 @@
+import { VizPanelBuilder } from '@grafana/scenes';
+import { PromQuery } from 'app/plugins/datasource/prometheus/types';
+
+export interface AutoQueryDef {
+  variant: string;
+  title: string;
+  unit: string;
+  queries: PromQuery[];
+  vizBuilder: (def: AutoQueryDef) => VizPanelBuilder<{}, {}>;
```

Comment:
think it would be nice to change this signature to just, was on my todo 
vizBuilder: () => VizPanelBuilder<{}, {}>;